src/mame/drivers/brother_ax145.cpp

```diff
@@ -0,0 +1,352 @@
+		dict_bank = data;
+	}
+	uint8_t dict_r(offs_t offset, uint8_t mem_mask = ~0) {
+		if(dict_bank >= 4) {
+			logerror("%s: illegal rombank (IO 5000=%02x) read offset %06x\n", pc(), dict_bank, offset);
+			return 0x00;
+		}
+		return dict_rom[offset + dict_bank * 0x20000] & mem_mask;
+	}
+
+	// int2
+	TIMER_DEVICE_CALLBACK_MEMBER(int2_timer_callback) {
+		maincpu->set_input_line(INPUT_LINE_IRQ2, ASSERT_LINE);
+	}
+	uint8_t irq_ack_r() {
+		maincpu->set_input_line(INPUT_LINE_IRQ2, CLEAR_LINE);
+		return 0;
+	}
+
+protected:
+	// driver_device overrides
+	void machine_start() override;
+	void machine_reset() override;
+
+	void video_start() override;
+
+	void map_program(address_map& map) {
+		map(0x00000, 0x01fff).rom();
+		map(0x04000, 0x1ffff).rom();
+		map(0x40000, 0x5ffff).r(FUNC(ax145_state::dict_r));
+		// RAM is installed in machine_start()
+	}
+
+	void map_io(address_map& map) {
+		map.global_mask(0xffff);
+		map(0x0000, 0x003f).noprw(); // Z180 internal registers
+		//map(0x0040, 0x00ff).rw(FUNC(ax145_state::illegal_io_r), FUNC(ax145_state::illegal_io_w));
+		//map(0x2000, 0x2000).w(TODO);
+		//map(0x2800, 0x2800).w(TODO);
+		map(0x3000, 0x3000).w(FUNC(ax145_state::io_3000_w));
+		map(0x3800, 0x3800).r(FUNC(ax145_state::io_3800_r)); // config lower 4 bits
+		map(0x4000, 0x4000).r(FUNC(ax145_state::io_4000_r)); // config upper 4 bits
+		map(0x5000, 0x5000).w(FUNC(ax145_state::io_5000_w)); // dictionary banking
+		map(0x5800, 0x5800).w(FUNC(ax145_state::lcd_signal_w));
+		map(0x6000, 0x6000).rw(FUNC(ax145_state::lcd_data_r), FUNC(ax145_state::lcd_data_w));
+		//map(0x6800, 0x6800).w(TODO);
+		//map(0x7000, 0x7000).w(TODO);
+		//map(0x7800, 0x7800).w(TODO);
+		//map(0x8000, 0x8000).w(TODO);
+		//map(0x8800, 0x8800).w(TODO);
+		//map(0x9000, 0x9000).r(TODO);
+		map(0xb000, 0xb000).r(FUNC(ax145_state::irq_ack_r));
+		//map(0xb800, 0xb800).rw(TODO, TODO);
+	}
+
+	void palette(palette_device &palette) const {
+		palette.set_pen_color(0, rgb_t(138, 146, 148));
+		palette.set_pen_color(1, rgb_t(92, 83, 88));
+	}
+
+	uint8_t* rom{};
+	uint8_t* dict_rom{};
+	std::map<uint32_t, std::string> symbols;
+};
+
+void ax145_state::video_start()
+{
+}
+
+std::string ax145_state::pc()
+{
+	class z180_friend : public z180_device { public: using z180_device::memory_translate; friend class ax145_state; };
+	auto cpu = static_cast<z180_friend*>(dynamic_cast<z180_device*>(&machine().scheduler().currently_executing()->device()));
+	offs_t phys = cpu->pc();
+	cpu->memory_translate(AS_PROGRAM, 0, phys);
+
+	return symbolize(phys);
+}
+
+std::string ax145_state::symbolize(uint32_t adr)
+{
+	if(symbols.empty())
+		return string_format("%06x", adr);
+
+	auto floor_it = symbols.lower_bound(adr);
+	if((floor_it == symbols.end() && !symbols.empty()) || floor_it->first != adr)
+		--floor_it;
+	if(floor_it != symbols.end())
+		return string_format("%s+%x (%06x)", floor_it->second, adr - floor_it->first, adr);
+	else
+		return string_format("%06x", adr);
+}
```

Comment:
Please use debugger extension features for whatever this is supposed to do.

Reply:
removed.